README.md

```diff
@@ -1,20 +1,88 @@
+|    P.1    |        Past Spork support for registers       |         X        |                   |                  |                   |    X    |      X     |             |        X       |                |                |                   |                    |              |
+|    P.2    |         Past Spork support with events        |         X        |         X         |                  |                   |    X    |      X     |             |        X       |                |                |                   |                    |              |
+|    R.1    |    Rosetta API support for default balance    |         X        |         X         |                  |                   |    X    |      X     |      X      |        X       |                |                |                   |                    |              |
+|    L.1    |        Live Spork support for registers       |         X        |         X         |         X        |                   |    X    |      X     |      X      |        X       |                |                |                   |                    |              |
+|    L.2    |         Live Spork support with events        |         X        |         X         |         X        |         X         |    X    |      X     |      X      |        X       |                |                |                   |                    |              |
+|    R.2    | Rosetta API support with sub-account balances |         X        |         X         |         X        |         X         |    X    |      X     |      X      |        X       |        X       |        X       |         X         |                    |              |
+|    C.1    |   Cryptographic Verification of local state   |         X        |         X         |         X        |         X         |    X    |      X     |      X      |        X       |        X       |        X       |         X         |          X         |              |
+|    C.2    |  Cryptographic Proofs for remote state access |         X        |         X         |         X        |         X         |    X    |      X     |      X      |        X       |        X       |        X       |         X         |          X         |       X      |
+
+## Architecture
+
+### Components
+
+The Flow Data Provisioning Service is constituted of three main components.
+
+The **indexer** is at the core of the service. It indexes deltas into its own data structures and provides a low level
+API to expose its state.
+The **mapper** synchronizes the **indexer** with a source of deltas (`FilesystemChain` or `NetworkChain`.)
+The **chain** is in charge of fetching block sequence data (from a WAL in the filesystem for past sporks, or a network
+socket for live ones.)
```

Comment:
```
The Flow Data Provisioning Service is organized into five different components, each with their own interface to accomplish their well-delineated responsibility.

1. The **Chain** interface is responsible for reconstructing a view of the sequence of blocks, along with their meta data. It allows the consumer to step from the root block to the last sealed block, while presenting height, block identifier and state commitment for each step. The file i/o version will do so by using the execution node's on-disk key-value store, while the network version will rely on data retrieved from access nodes.
2. The **Streamer** interface is responsible for streaming in-order trie updates from different sources; the file i/o version will read them from the LedgerWAL, while the network version will receive trie updates through its network subscription on the execution node.
3. The **Mapper** interface is responsible for mapping incoming state trie updates to blocks. Generally, trie updates come in by chunk, so each block will map from zero to multiple trie updates. Once a block is mapped to its respective trie updates, the mapper forwards the information to the indexer.
4. The **Indexer** interface is responsible for receiving a set of trie updates for each block and creating the necessary main indexes and auxiliary in the on-disk database. These indexes will allow efficient retrieval of the state at arbitrary block heights in the state history.
5. The **Ledger** interface is responsible for providing clean APIs to access the execution state at different block heights and registers. They use the underlying indexes to emulate the same APIs as the Flow execution node Ledger, or the Rosetta Data API.
```

Reply:
```suggestion
The Flow Data Provisioning Service is constituted of three main components.

1. The **Chain** interface is responsible for reconstructing a view of the sequence of blocks, along with their meta data. It allows the consumer to step from the root block to the last sealed block, while presenting height, block identifier and state commitment for each step. The file i/o version does so by using the execution node's on-disk key-value store, while the network version relies on data retrieved from access nodes.
2. The **Streamer** interface is responsible for streaming in-order trie updates from different sources; the file i/o version reads them from the LedgerWAL, while the network version receives trie updates through its network subscription on the execution node.
3. The **Mapper** interface is responsible for mapping incoming state trie updates to blocks. Generally, trie updates come in by chunk, so each block maps from zero to multiple trie updates. Once a block is mapped to its respective trie updates, the mapper forwards the information to the indexer.
4. The **Indexer** interface is responsible for receiving a set of trie updates for each block and creating the necessary main indexes and auxiliary in the on-disk database. These indexes allow efficient retrieval of the state at arbitrary block heights in the state history.
5. The **Ledger** interface is responsible for providing clean APIs to access the execution state at different block heights and registers. Its implementations use the underlying indexes to emulate the same APIs as the Flow execution node Ledger, or the Rosetta Data API.
```

WDYT of getting rid of the future tense here too? Also on the last part, I replaced `They use` which I assumed meant `The implementations of the Ledger interface` with `Its implementations`. SGTY?